Add nn.quantized.Conv3d

xiaomengy opened the pull request:
Summary: Add nn.quantized.Conv3d

Test Plan: buck test mode/dev-nosan //caffe2/test:quantized -- "conv"

Reviewed By: jianyuh

Differential Revision: D18467749

BIT-silence commented:
Link to https://github.com/pytorch/pytorch/issues/28480

jianyuh commented:
Thanks for working on this. Overall it looks good. Check the OSS lint issues such as
```
./test/test_quantized_nn_mods.py:11:1: F401 'torch.nn.quantized.modules.Conv2d' imported but unused
./test/test_quantized_nn_mods.py:12:1: F401 'torch.nn.intrinsic.quantized.ConvReLU2d' imported but unused
```

BIT-silence commented:
> Thanks for working on this. Overall it looks good. Check the OSS lint issues such as
> 
> ```
> ./test/test_quantized_nn_mods.py:11:1: F401 'torch.nn.quantized.modules.Conv2d' imported but unused
> ./test/test_quantized_nn_mods.py:12:1: F401 'torch.nn.intrinsic.quantized.ConvReLU2d' imported but unused
> ```

Done

houseroad reviewed (approved):
Looks good.

houseroad commented on torch/nn/intrinsic/quantized/modules/conv_relu.py:
```diff
@@ -18,30 +21,70 @@ class ConvReLU2d(nnq.Conv2d):
+            padding_mode=padding_mode)
+
+    def forward(self, input):
+        # Temporarily using len(shape) instead of ndim due to JIT issue
```
Add a TODO?

houseroad commented on test/test_quantized_nn_mods.py:
```diff
@@ -28,26 +27,26 @@
-    batch_size, input_channels_per_group, input_feature_map_size,
-    output_channels_per_group, groups, kernel_size, X_scale, X_zero_point,
-    W_scale, W_zero_point, use_bias, use_channelwise,
+    batch_size, in_channels_per_group, input_feature_map_size,
```
input_feature_map_size => in_feature_map_size?

xiaomengy committed:
Add nn.quantized.Conv3d (#29813)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/29813

Add nn.quantized.Conv3d

Test Plan: buck test mode/dev-nosan //caffe2/test:quantized -- "conv"

Reviewed By: jianyuh

Differential Revision: D18467749

fbshipit-source-id: 3f992de8f4100f3bd7ba696a6e8943d027d8b2f8
